feat: custom UI for autocomplete place

theLimitBreaker opened the pull request:
Fixes #1529 

Changes:
- Add geocoding API

Current Status:
- results are fetched for autocomplete place suggesstions

iamareebjamal commented on app/src/main/java/org/fossasia/openevent/general/di/Modules.kt:
```diff
@@ -193,6 +199,16 @@ val networkModule = module {
             .baseUrl(baseUrl)
             .build()
     }
+    single<Retrofit>("autoPlace_api") {
```
Why not using geocoding jar from mapbox?

theLimitBreaker replied:
Do you mean extracting jar from MapBox Sdk(that added for Autocomplete) ?

theLimitBreaker replied:
Since last time adding Sdk introduced too much size
I thought too use Api request method as we only need the list of autocomplete places . We would have the full control over code 
Plus we wont have no extra methods and size of MapBox Java Sdk

theLimitBreaker replied:
These are api provided with Mapbox java sdk services https://docs.mapbox.com/android/api/mapbox-java/libjava-services/3.0.1/index.html 

I think I can't select or exclude other packages in services dependency.

iamareebjamal replied:
Use pure java lib of MapBox geocoding. We don't want to maintain API classes of MapBox. Implement your own solution only if the jar contains mapbox-gl-native, which I suppose it won't

theLimitBreaker replied:
I understand. I checked doesn't include mapbox-gl-native
proceeding with Mapbox java sdk then

theLimitBreaker commented:
### Update 
 **Current Status**
- removed Manual handling of API,
- Added Java Sdk Mapbox
- Displaying results of autocomplete in recyclerview

<img src="https://user-images.githubusercontent.com/24780524/55670387-35606700-58a1-11e9-912a-64d521b822a8.gif" width=360>

liveHarshit commented:
build is failing

theLimitBreaker commented:
**Update**

<img src="https://user-images.githubusercontent.com/24780524/55850274-4941ec80-5b71-11e9-90ed-b285fbb7fa53.gif" width=360>

liveHarshit commented:
> **Update**
> 

same

theLimitBreaker commented:
> Task : TransformClassesForfdroidReleas is failing mainly . Trying to figure out the reason

aggarwalpulkit596 commented:
@theLimitBreaker still working on this ?

theLimitBreaker commented:
Yes ! It's WIP PR .

theLimitBreaker commented:
@liveHarshit  could you help me with this 

```
Warning: there were 171 unresolved references to classes or interfaces.
         You may need to add missing library jars or update their versions.
         If your code works fine without the missing classes, you can suppress
         the warnings with '-dontwarn' options.
         (http://proguard.sourceforge.net/manual/troubleshooting.html#unresolvedclass)
Warning: Exception while processing task java.io.IOException: Please correct the above warnings first.
Thread(Tasks limiter_1): destruction

> Task :app:transformClassesAndResourcesWithProguardForFdroidRelease FAILED

FAILURE: Build failed with an exception.
```

iamareebjamal commented:
See mapbox android project and their proguard rules

theLimitBreaker commented:
**Pre Final Update**
- Hide Popular Places when PlaceSuggestions Come and Vice-Versa (Like Event Brite)
- add Debounce to make place-suggestion query (to avoid unnecessary calls)
- refractor code (to make code base clean)

<img src="https://user-images.githubusercontent.com/24780524/56008964-b3869880-5cfb-11e9-9af4-08d428f5c8c5.gif" width=360>

theLimitBreaker commented:
@iamareebjamal I have checked proguard-rules.pro of `mapbox-android-demo`. They have added 
`-dontwarn` infront of specific mapbox classes/libraries unlike `com.mapbox.**`

theLimitBreaker commented:
@ShridharGoel  @liveHarshit   @iamareebjamal could you help me a bit with squashing my commits . I made some mistake while rebasing and squashing my own commits
